webpack/assets/javascripts/react_app/components/common/forms/FormField.js

```diff
@@ -0,0 +1,140 @@
+import { omit } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Col,
+  FormGroup,
+  FormControl,
+  ControlLabel,
+  HelpBlock,
+  FieldLevelHelp,
+} from 'patternfly-react';
+
+import { noop } from '../../../common/helpers';
+import DateTimePicker from '../DateTimePicker/DateTimePicker';
+import DatePicker from '../DateTimePicker/DatePicker';
+import TimePicker from '../DateTimePicker/TimePicker';
+
+const inputComponents = {
+  date: DatePicker,
+  dateTime: DateTimePicker,
+  time: TimePicker,
+};
+const componentTypes = Object.keys(inputComponents);
+const inputTypes = componentTypes.concat(['text']);
+
+const FormField = ({
+  type,
+  id,
+  name,
+  className,
+  disabled,
+  required,
+  touched,
+  error,
+  value,
+  label,
+  labelHelp,
+  labelSizeClass,
+  inputSizeClass,
+  onChange,
+  ...props
+}) => {
+  const validationState = touched && error ? 'error' : null;
+
+  return (
+    <FormGroup
+      controlId={id}
+      disabled={disabled}
+      validationState={validationState}
+    >
+      <ControlLabel className={labelSizeClass}>
+        {label}
+        {labelHelp && (
+          <FieldLevelHelp
+            placement="right"
+            buttonClass="field-help"
+            content={<React.Fragment>{labelHelp}</React.Fragment>}
+          />
+        )}
+        {required && '*'}
+      </ControlLabel>
+      <Col className={inputSizeClass}>
+        <FormControl
+          componentClass={inputComponents[type]}
```

Comment:
Note that this restricts form fields to those that are in core. Plugins will not be able to use their own input components if they need [something special](https://github.com/theforeman/foreman_templates/pull/109/files#diff-5f4a0359daf2426a6b0874b78886c28cR31).

Reply:
Well, that is not really true, we would just have to figure a way for them to be able to register new type, what should not be so hard to do.

Reply:
Ok, I have done both aproaches. You can choose now. Maybe I could add even one - higher order component wrapping any field up nicely?

Reply:
Registering would work, but I do not see what benefits it brings us when passing children achieves the same and has no downsides as far as I can see. But maybe that is just a personal preference.

Reply:
The advantage is simply in not exposing all of the children in `componentRegistry` and still be able to reach them from Rails. Like this, we have only one mount point :)